internal/syncd/integrations/ecr/reconciler.go

```diff
@@ -2,27 +2,46 @@ package ecr
 
 import (
 	"context"
-
+	"fmt"
+	"path/filepath"
 	"ship-it/internal"
 
-	"github.com/google/go-github/github"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/types"
 )
 
 type RepositoriesService interface {
-	GetContents(ctx context.Context, org, repo, path string, opts *github.RepositoryContentGetOptions) (*github.RepositoryContent, []*github.RepositoryContent, *github.Response, error)
-	UpdateFile(ctx context.Context, owner, repo, path string, opt *github.RepositoryContentFileOptions) (*github.RepositoryContentResponse, *github.Response, error)
+	UpdateAndReplace(ctx context.Context, path string, image *internal.Image, msg string) error
+}
+
+type IndexerService interface {
+	Lookup(repo string) ([]types.NamespacedName, error)
 }
 
 type ImageReconciler struct {
-	// TODO
+	RegistryChartPath string
+	RepoService       RepositoriesService
+	IndexService      IndexerService
 }
 
-func NewReconciler(org string, r RepositoriesService) *ImageReconciler {
+func NewReconciler(prefix string, r RepositoriesService, i IndexerService) *ImageReconciler {
 	return &ImageReconciler{
-		// TODO
+		RegistryChartPath: prefix,
```

Comment:
The `RegistryChartPath` field exists only to be passed to the `RepoService.UpdateAndReplace` method call. This suggests that the field should belong to the `RepoService` instead.

I think the `RepositoriesService` interface name should be updated as well, since the old one was a reference to go-github's [RepositoriesService](https://godoc.org/github.com/google/go-github/github#RepositoriesService). Something like `HelmReleaseEditor` or `HelmReleaseUpdater`  would make sense to me

Reply:
Renamed to `HelmReleaseEditor`